jiant/modules/seq2seq_decoder.py

```diff
@@ -185,10 +191,15 @@ def forward(
         output_dict = {"logits": logits}
 
         if target_tokens:
+            targets = target_tokens["words"]
             target_mask = get_text_field_mask(target_tokens)
-            loss = self._get_loss(logits, targets, target_mask)
-            output_dict["loss"] = loss
             output_dict["target_mask"] = target_mask
+            relevant_logits = logits[:, : targets.shape[1] - 1, :].contiguous()
+            loss = self._get_loss(relevant_logits, targets, target_mask)
+            output_dict["loss"] = loss
+        else:
+            # This is needed in case no gold target sequence is available.
+            output_dict["loss"] = torch.tensor([-1.0])
```

Comment:
fatal to use `None`?

Reply:
I think I tried that, yes. But it will also be changed in a follow-up PR.